lib/network/api/transaction_test.go

```diff
@@ -0,0 +1,212 @@
+	ts, storage, err := prepareAPIServer()
+	require.Nil(t, err)
+	defer storage.Close()
+	defer ts.Close()
+
+	btList, err := prepareTxs(storage, 0, 10)
+	require.Nil(t, err)
+
+	// streaming
+	{
+		// Do a Request
+		respBody, err := request(ts, GetTransactionsHandlerPattern, true)
+		require.Nil(t, err)
+		defer respBody.Close()
+		reader := bufio.NewReader(respBody)
+
+		// Producer
+		recv := make(chan struct{})
+		go func() {
+			btList2, err := prepareTxs(storage, 1, 10)
+			require.Nil(t, err)
+			btList = append(btList, btList2...)
+			close(recv)
+		}()
+		var btMap = make(map[string]block.BlockTransaction)
+		for _, bt := range btList {
+			btMap[bt.Hash] = bt
+		}
+
+		// Do stream Request to the Server
+		<-recv
+		for n := 0; n < 10; n++ {
+			line, err := reader.ReadBytes('\n')
+			require.Nil(t, err)
+			line = bytes.Trim(line, "\n\t ")
+			var receivedBt block.BlockTransaction
+			json.Unmarshal(line, &receivedBt)
+			txS, err := btMap[receivedBt.Hash].Serialize()
+			require.Nil(t, err)
+			require.Equal(t, txS, line)
+		}
+	}
+}
+
+func TestGetTransactionsHandlerPage(t *testing.T) {
+	ts, storage, err := prepareAPIServer()
+	require.Nil(t, err)
+	defer storage.Close()
+	defer ts.Close()
+
+	btList, err := prepareTxs(storage, 0, 10)
+	require.Nil(t, err)
+
+	testFunction := func (query string) []block.BlockTransaction{
+		respBody, err := request(ts, GetTransactionsHandlerPattern + "?" + query, false)
+		require.Nil(t, err)
+		defer respBody.Close()
+		reader := bufio.NewReader(respBody)
+
+		readByte, err := ioutil.ReadAll(reader)
+		require.Nil(t, err)
+		var receivedBts []block.BlockTransaction
+		json.Unmarshal(readByte, &receivedBts)
+		return receivedBts
+	}
+	{
+		query := strings.Replace(QueryPattern, "{cursor}", "", 1)
+		query = strings.Replace(query, "{limit}", "0", 1)
+		query = strings.Replace(query, "{reverse}", "false", 1)
+		receivedBts := testFunction(query)
+		require.Equal(t, len(btList), len(receivedBts), "length is not same")
+
+		for i, bt := range receivedBts {
+			require.Equal(t, bt.Hash, btList[i].Hash, "hash is not same")
+		}
+	}
+	{
+		query := strings.Replace(QueryPattern, "{cursor}", "", 1)
+		query = strings.Replace(query, "{limit}", "5", 1)
+		query = strings.Replace(query, "{reverse}", "false", 1)
+		receivedBts := testFunction(query)
+		require.Equal(t, len(btList[:5]), len(receivedBts), "length is not same")
+
+		for i, bt := range receivedBts {
+			require.Equal(t, bt.Hash, btList[i].Hash, "hash is not same")
+		}
+	}
+	{
+		query := strings.Replace(QueryPattern, "{cursor}", "", 1)
+		query = strings.Replace(query, "{limit}", "0", 1)
+		query = strings.Replace(query, "{reverse}", "true", 1)
+		receivedBts := testFunction(query)
+		require.Equal(t, len(btList), len(receivedBts), "length is not same")
+
+		for i, bt := range receivedBts {
+			require.Equal(t, bt.Hash, btList[len(btList) - 1 - i].Hash, "hash is not same")
+		}
+	}
+
+	//TODO: cursor
```

Comment:
for the cursor, we need the response format to parse. @anarcher

Reply:
I am doing to apply HAL to handlers #363 :->